marketplace-service/src/main/java/com/axonivy/market/github/service/impl/GitHubServiceImpl.java

```diff
@@ -172,12 +166,35 @@ public void validateUserInOrganizationAndTeam(String accessToken, String organiz
     }
 
     throw new UnauthorizedException(ErrorCode.GITHUB_USER_UNAUTHORIZED.getCode(),
-        String.format(ErrorMessageConstants.INVALID_USER_ERROR, ErrorCode.GITHUB_USER_UNAUTHORIZED.getHelpText(),
-            team, organization));
+        String.format(ErrorMessageConstants.INVALID_USER_ERROR, ErrorCode.GITHUB_USER_UNAUTHORIZED.getHelpText(), team,
+            organization));
   }
 
-  private boolean isUserInOrganizationAndTeam(GitHub gitHub, String organization,
-      String teamName) throws IOException {
+  @Override
+  public List<ProductSecurityInfo> getSecurityDetailsForAllProducts(String accessToken, String orgName) {
+    List<ProductSecurityInfo> securityInfoList;
+    ExecutorService executor = taskScheduler.getScheduledExecutor();
+    try {
+      GitHub gitHub = getGitHub(accessToken);
+      GHOrganization organization = gitHub.getOrganization(orgName);
+
+      List<CompletableFuture<ProductSecurityInfo>> futures = organization.listRepositories().toList().stream()
+          .map(repo -> CompletableFuture.supplyAsync(() -> fetchSecurityInfoSafe(repo, organization, accessToken), executor))
+          .toList();
+
+      securityInfoList = futures.stream()
+          .map(CompletableFuture::join)
+          .collect(Collectors.toList());
+
+      securityInfoList.sort(Comparator.comparing(ProductSecurityInfo::getRepoName));
+    } catch (IOException e) {
+      throw new RuntimeException("Error fetching repository data", e);
```

Comment:
We should log it, and return an empty list when the system cannot fetch a repository